### DISCUSSION TITLE
How To Configure Login Link (Forgot Password) ASAP

### DISCUSSION THREAD

**Discussion by kodetumbuh** [codeigniter4/shield#569] [Q&A] [answered]
i've been visit this page [](https://codeigniter4.github.io/shield/install/) really work and good, however can't find how to configure Login Link (forgot password). i try configure a file in app/config/email and try other ways but still work could you help how to configure login link or atleast a tutorial thanks

![image](https://user-images.githubusercontent.com/46043848/209454291-603cd7b9-7da8-4df8-8141-88b727a41633.png)

**Comment by kenjis** [comment]
What do you want to do exaclty?
I don't understand *Configure Login Link (Forgot Password)* means,
and why you put the screenshot.

**Reply by kodetumbuh** [reply]
![image](https://user-images.githubusercontent.com/46043848/209454405-60904c35-1efb-40d3-8080-7dda9fad1004.png)
sorry english is not my native languange, i mean how configure this page right here 

http://localhost:8080/index.php/login/magic-link

**Reply by datamweb** [reply]
Everything is already set up, your error is about not setting up file `app/Config/Email.php`.
see #289

**Reply by kenjis** [reply]
> sorry english is not my native languange

No problem! English is not my native language.

**Comment by kodetumbuh** [comment, marked as answer]
i found the problem for gmail (you can't use old password need an app to use it)
https://support.google.com/accounts/answer/185833

how to setting Use login link (forgot password) :
1. make new password here https://support.google.com/accounts/answer/185833
2. setting smtp like this 
```php
public $fromEmail = '*****@gmail.com';
public $fromName = 'your name';
public $protocol = 'smtp';
public $SMTPHost = 'smtp.googlemail.com';
public $SMTPUser = '****@gmail.com';
public $SMTPPass = '***(insert your an app gmail here)';
public $SMTPPort = 465;
public $SMTPCrypto = 'ssl';
```
3.done and test it

i think to make clear documentation shield, add a tutorial how to setting email that's will be helpfull especially with begineer like me. thank you kenjis and datamweb for your support!
